ddcommon/src/lib.rs

```diff
@@ -82,19 +89,22 @@ fn deserialize_uri<'de, D>(deserializer: D) -> Result<hyper::Uri, D::Error>
 where
     D: Deserializer<'de>,
 {
-    let uri = SerializedUri::deserialize(deserializer)?;
-    let mut builder = hyper::Uri::builder();
-    if let Some(v) = uri.authority {
-        builder = builder.authority(v.deref());
-    }
-    if let Some(v) = uri.scheme {
-        builder = builder.scheme(v.deref());
-    }
-    if let Some(v) = uri.path_and_query {
-        builder = builder.path_and_query(v.deref());
+    match StringOrSerializedUri::deserialize(deserializer)? {
+        StringOrSerializedUri::String(str) => str.parse().map_err(Error::custom),
```

Comment:
are there ways to add a message here to differentiate between the errors ?

Reply:
I think the original error message will be properly remapped onto the new custom error, or at least that is my understanding of what this is doing"